File: Python/Program/Class/Motherboard.py
from Class.PartKomputer import PartKomputer

class Motherboard(PartKomputer):
    def __init__(self, id=0, nama="", harga=0, stok=0,
                 no_part="", manufaktur="", kondisi="",
                 socket="", ddr_ver=0, chipset=""):
        super().__init__(id, nama, harga, stok, no_part, manufaktur, kondisi)
        self._socket = socket
        self._ddr_ver = ddr_ver
        self._chipset = chipset

    def set_socket(self, socket):
        self._socket = socket

    def get_socket(self):
        return self._socket

    def set_ddr_ver(self, ddr_ver):
        self._ddr_ver = ddr_ver

    def get_ddr_ver(self):
        return self._ddr_ver

    def set_chipset(self, chipset):
        self._chipset = chipset

    def get_chipset(self):
        return self._chipset

File: Java/Program/Class/Motherboard.java
package Class;

public class Motherboard extends PartKomputer {
    private String socket;
    private int DDR_Ver;
    private String chipset;

    public Motherboard() {
        super();
        this.socket = "";
        this.DDR_Ver = 0;
        this.chipset = "";
    }

    public Motherboard(int id, String nama, int harga, int stok,
                       String noPart, String manufaktur, String kondisi,
                       String socket, int DDR_Ver, String chipset) {
        super(id, nama, harga, stok, noPart, manufaktur, kondisi);
        this.socket = socket;
        this.DDR_Ver = DDR_Ver;
        this.chipset = chipset;
    }

    public String getSocket() { return socket; }
    public void setSocket(String socket) { this.socket = socket; }

    public int getDDRVer() { return DDR_Ver; }
    public void setDDRVer(int DDR_Ver) { this.DDR_Ver = DDR_Ver; }

    public String getChipset() { return chipset; }
    public void setChipset(String chipset) { this.chipset = chipset; }
}
